src/firewall/core/ipXtables.py

```diff
@@ -1116,15 +1116,22 @@ def _rich_rule_destination_fragment(self, rich_dest):
             return []
 
         rule_fragment = []
-        if rich_dest.invert:
-            rule_fragment.append("!")
-        if check_single_address("ipv6", rich_dest.addr):
-            rule_fragment += [ "-d", normalizeIP6(rich_dest.addr) ]
-        elif check_address("ipv6", rich_dest.addr):
-            addr_split = rich_dest.addr.split("/")
-            rule_fragment += [ "-d", normalizeIP6(addr_split[0]) + "/" + addr_split[1] ]
-        else:
-            rule_fragment += [ "-d", rich_dest.addr ]
+        if rich_dest.addr:
+            if rich_dest.invert:
+                rule_fragment.append("!")
+            if check_single_address("ipv6", rich_dest.addr):
+                rule_fragment += [ "-d", normalizeIP6(rich_dest.addr) ]
+            elif check_address("ipv6", rich_dest.addr):
+                addr_split = rich_dest.addr.split("/")
+                rule_fragment += [ "-d", normalizeIP6(addr_split[0]) + "/" + addr_split[1] ]
+            else:
+                rule_fragment += [ "-d", rich_dest.addr ]
+        elif hasattr(rich_dest, "ipset") and rich_dest.ipset:
```

Comment:
nit: `hasattr()` in unnecessary as you guarantee `ipset` exists when creating the `Rich_Destination` instance.